simapp/app_v2.go

```diff
@@ -281,8 +274,12 @@ func NewSimApp(
 
 	app.sm.RegisterStoreDecoders()
 
-	// initialize BaseApp
-	app.SetInitChainer(app.InitChainer)
+	// Set custom init chainer that registers the module versions in the upgrade module
+	// NOTE: this is not required for apps that do not need custom pre-init genesis logic
+	app.SetInitChainer(func(ctx sdk.Context, req abci.RequestInitChain) abci.ResponseInitChain {
+		app.UpgradeKeeper.SetModuleVersionMap(ctx, app.ModuleManager.GetVersionMap())
+		return app.App.InitChainer(ctx, req)
```

Comment:
Why can't we have this get called automatically by runtime? Is there any scenario we'd want to not call the default init genesis functionality?

Reply:
It is for avoiding creating a new API that we might remove when we improve app wiring to support this use case.

Now we can safely delete SetInitChainer from runtime and add the equivalent to app wiring when we figured out how.

We can indeed simplify the function but then when we remove it it's API breaking.

Reply:
Updated